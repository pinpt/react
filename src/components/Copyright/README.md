### Displays copyright text for your site, either with or without a logo

[Full Docs](pinpt.github.io/react/?path=/docs/components-copyright) | [Github](https://github.com/pinpt/react/tree/master/src/components/Copyright)

#### Theming

The parent className for styling this component is `.Copyright`

<details>
	<summary>Base Theme Styles</summary>

```css
.Copyright {
	line-height: 1.5;
}

.Copyright .text {
	margin-top: 1rem;
	font-size: 0.875rem;
}
```

</details>
