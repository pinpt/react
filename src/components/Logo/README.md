### Displays your site's logo. Can be made into a link or an onClick handler can be attached.

[Full Docs](https://react.preview.pinpoint.com/?path=/docs/components-logo) | [Github](https://github.com/pinpt/react/tree/master/src/components/Logo)

#### Theming

The parent className for styling this component is `.Logo`

<details>
	<summary>Base Theme Styles</summary>

```css
.Logo {
	@apply block rounded-full overflow-hidden;
}

.Logo.xs {
	width: 22px;
	height: 22px;
}

.Logo.sm {
	width: 32px;
	height: 32px;
}

.Logo.md {
	width: 42px;
	height: 42px;
}

.Logo.lg {
	width: 52px;
	height: 52px;
}

.Logo.xl {
	width: 62px;
	height: 62px;
}

.Logo .image {
	width: 100%;
	height: 100%;
}
```

</details>
