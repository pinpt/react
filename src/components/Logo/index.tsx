import ActionLink from '../Internal/ActionLink';
import React from 'react';
export interface ILogoProps {
	className?: string;
	src?: string;
	href?: string;
	newTab?: boolean;
	size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
	onClick?: () => void;
	title?: string;
}

const Logo = (props: ILogoProps) => {
	const { className = '', src, href, newTab, size = 'sm', onClick, title = 'Logo' } = props;

	return (
		<ActionLink
			className={`Pinpoint Logo ${size} ${className}`}
			href={href}
			newTab={newTab}
			onClick={onClick}
			title={title}
		>
			<img className="image" src={src} />
		</ActionLink>
	);
};

export default Logo;
