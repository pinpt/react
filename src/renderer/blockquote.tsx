import { registerNode, recurseIntoChildren, NodeProps } from './register';

const BlockQuote = ({ node }: NodeProps) => {
	return <blockquote>{recurseIntoChildren(node)}</blockquote>;
};

registerNode('blockquote', (node) => <BlockQuote node={node} />);
