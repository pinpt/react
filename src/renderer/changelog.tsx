import { ReactYouTubeLite as Youtube } from 'react-youtube-lite';
import { getFileUrl } from '../lib/file_metadata';
import { slugifyContent } from '../lib/string';
import { ChangelogCoverMedia, ChangelogMediaType } from '../lib/types';
import { Content as Doc } from './';

const ImageMedia = ({ src, title = '' }: { src: string; title?: string }) => {
	return (
		<div className="image">
			<img src={src} alt={title} />
		</div>
	);
};

const VideoMedia = ({ src }: { src: string }) => {
	return (
		<div className="video">
			<video controls src={src} />
		</div>
	);
};

const YoutubeMediaSSR = ({ id, poster = 'hqdefault' }: { id: string; poster?: string }) => {
	// the component isn't SSR friendly so this is basically used to sort of mimic it for generated code
	const posterUrl = `https://i.ytimg.com/vi/${id}/${poster}.jpg`;
	return (
		<>
			{poster && <link rel="preload" href={posterUrl} as="image" />}
			<div
				className="youtube ryt-lite embed-responsive aspect-ratio-16/9"
				style={{ backgroundImage: `url(${posterUrl})` }}
			>
				<div className="lty-playbtn" data-url={`https://www.youtube-nocookie.com/embed/${id}?autoplay=1`} />
			</div>
		</>
	);
};

const YoutubeMedia = ({
	id,
	metadata,
	staticMode,
}: {
	id: string;
	metadata?: Record<string, any>;
	staticMode?: true;
}) => {
	return (
		<div className="youtube">
			{!staticMode ? (
				<Youtube url={`https://www.youtube.com/watch?v=${id}`} poster={metadata?.poster} />
			) : (
				<YoutubeMediaSSR id={id} poster={metadata?.poster} />
			)}
		</div>
	);
};

const Media = ({ media, title, staticMode }: { media?: ChangelogCoverMedia; title?: string; staticMode?: true }) => {
	if (!media) {
		return <></>;
	}
	let content: React.ReactElement;
	switch (media.type) {
		case ChangelogMediaType.None: {
			return <></>;
		}
		case ChangelogMediaType.Image: {
			content = <ImageMedia src={getFileUrl(media.value)} title={title} />;
			break;
		}
		case ChangelogMediaType.Video: {
			content = <VideoMedia src={getFileUrl(media.value)} />;
			break;
		}
		case ChangelogMediaType.Youtube: {
			content = <YoutubeMedia id={media.value} metadata={media.metadata} staticMode={staticMode} />;
			break;
		}
	}
	return <div className="media-container">{content}</div>;
};

interface ChangelogProps {
	id: string;
	title?: string;
	content?: any;
	coverMedia?: ChangelogCoverMedia;
	limit?: number;
	divider?: boolean;
	staticMode?: true;
}

const ChangelogLog = (props: ChangelogProps) => {
	return (
		<article className="changelog notebook-editor read-only">
			{props.coverMedia && (
				<section className="covermedia">
					<Media media={props.coverMedia} title={props.title} staticMode={props.staticMode} />
				</section>
			)}
			<section className="ProseMirror">
				<Doc node={props.content} limit={props.limit} />
				{props.limit && props.content.content?.length > props.limit && (
					<div className="continue">
						<a href={slugifyContent(props.id, props.title)}>Continue Reading â†’</a>
					</div>
				)}
			</section>
			{props.divider && <hr className="divider" />}
		</article>
	);
};

export default ChangelogLog;
