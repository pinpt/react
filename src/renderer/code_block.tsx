import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
import { twilight } from 'react-syntax-highlighter/dist/cjs/styles/prism';
import { NodeProps, registerNode } from './register';

const CodeBlock = ({ node }: NodeProps) => {
	return (
		<SyntaxHighlighter
			language={node.attrs.language}
			style={twilight}
			customStyle={{
				background: 'black',
				fontFamily: 'ui-monospace',
				borderWidth: '0px',
				borderRadius: '.25em',
			}}
		>
			{node.content?.[0].text ?? ''}
		</SyntaxHighlighter>
	);
};

registerNode('code_block', (node) => <CodeBlock node={node} />);
