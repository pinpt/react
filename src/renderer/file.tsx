import {
	faDownload, faFile, faFileAlt, faFileArchive, faFileAudio, faFileCode, faFileExcel, faFilePdf,
	faFilePowerpoint, faFileVideo, faFileWord
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { NodeProps, registerNode } from './register';

export const iconForType = (type = '') => {
	if (/pdf/.test(type)) {
		return faFilePdf;
	}
	if (/zip|tar/.test(type)) {
		return faFileArchive;
	}
	if (/video/.test(type)) {
		return faFileVideo;
	}
	if (/audio/.test(type)) {
		return faFileAudio;
	}
	if (/excel|spreadsheet/.test(type)) {
		return faFileExcel;
	}
	if (/powerpoint/.test(type)) {
		return faFilePowerpoint;
	}
	if (/word/.test(type)) {
		return faFileWord;
	}
	if (/css|csv|graphql|html|java|json|php|python|typescript|xml/.test(type)) {
		return faFileCode;
	}
	if (/text/.test(type)) {
		return faFileAlt;
	}
	return faFile;
};

export const formattedSize = (size = 0) => {
	const kB = size / 1000;
	if (kB < 10) {
		return `${Math.round(kB * 10) / 10} KB`;
	}
	if (kB < 1000) {
		return `${Math.round(kB)} KB`;
	}
	const MB = kB / 1000;
	return `${Math.round(MB * 10) / 10} MB`;
};

const File = ({ node }: NodeProps) => {
	const { description, filename, publicUrl, size, type } = node.attrs;

	const onDownload = () => {
		if (publicUrl) {
			const u = new URL(publicUrl);
			u.searchParams.set('filename', filename);
			u.searchParams.set('disposition', 'attachment');
			const a = document.createElement('a');
			a.style.display = 'none';
			a.href = u.toString();
			// the filename you want
			a.download = filename;
			document.body.appendChild(a);
			a.click();
			a.remove();
		}
	};

	return (
		<div className="file">
			<div className="container">
				<FontAwesomeIcon icon={iconForType(type)} />
				<div className="details">
					{publicUrl ? (
						<a href={publicUrl} target="_blank" rel="noopener noreferrer" className="filename">
							{filename}
						</a>
					) : (
						<div className="filename">{filename}</div>
					)}
					{description && <div className="description">{description}</div>}
					<div className="size">{formattedSize(size)}</div>
				</div>
				<div className="options">
					{publicUrl && (
						<div className="option group" onClick={onDownload}>
							<FontAwesomeIcon icon={faDownload} />
							<span>Download</span>
						</div>
					)}
				</div>
			</div>
		</div>
	);
};

registerNode('file', (node) => <File node={node} />);
