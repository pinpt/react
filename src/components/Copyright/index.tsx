import { ReactElement } from 'react';
import { ILogoProps } from '../Logo';

export interface ICopyrightProps {
	className?: string;
	text: string;
	logo?: ReactElement<ILogoProps>;
}

const Logo = (props: ICopyrightProps) => {
	const { className = '', text, logo } = props;
	return (
		<div className={`Pinpoint Copyright ${className}`}>
			{logo}
			<span className="text">Copyright Â© {text}</span>
		</div>
	);
};

export default Logo;
