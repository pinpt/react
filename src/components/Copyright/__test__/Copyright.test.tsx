import renderer from 'react-test-renderer';
import Copyright from '../';
import Logo from '../../Logo';

const IMAGE_URL =
	'https://file.pinpoint.com/1fcde4196a4c70a8a86f0ce4af53f2a5;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png';

test('Test with logo', () => {
	const component = renderer.create(
		<Copyright text="Copyright © 2021 by Pinpoint Software, Inc." logo={<Logo src={IMAGE_URL} />} />
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test without logo', () => {
	const component = renderer.create(<Copyright text="Copyright © 2021 by Pinpoint Software, Inc." />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test with clickable logo', () => {
	const component = renderer.create(
		<Copyright
			text="Copyright © 2021 by Pinpoint Software, Inc."
			logo={<Logo src={IMAGE_URL} onClick={() => console.log('click!')} />}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test with linked logo', () => {
	const component = renderer.create(
		<Copyright
			text="Copyright © 2021 by Pinpoint Software, Inc."
			logo={<Logo src={IMAGE_URL} href="https://pinpoint.com" />}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test with custom className', () => {
	const component = renderer.create(
		<Copyright
			text="Copyright © 2021 by Pinpoint Software, Inc."
			logo={<Logo src={IMAGE_URL} />}
			className="test-custom"
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
