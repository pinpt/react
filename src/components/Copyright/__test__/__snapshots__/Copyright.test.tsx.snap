// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test with clickable logo 1`] = `
<div
  className="Pinpoint Copyright "
>
  <span
    className="Pinpoint Logo sm clickable "
    onClick={[Function]}
  >
    <img
      className="image"
      src="https://file.pinpoint.com/1fcde4196a4c70a8a86f0ce4af53f2a5;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png"
    />
  </span>
  <span
    className="text"
  >
    Copyright © 
    2021 by Pinpoint Software, Inc.
  </span>
</div>
`;

exports[`Test with custom className 1`] = `
<div
  className="Pinpoint Copyright test-custom"
>
  <span
    className="Pinpoint Logo sm  "
  >
    <img
      className="image"
      src="https://file.pinpoint.com/1fcde4196a4c70a8a86f0ce4af53f2a5;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png"
    />
  </span>
  <span
    className="text"
  >
    Copyright © 
    2021 by Pinpoint Software, Inc.
  </span>
</div>
`;

exports[`Test with linked logo 1`] = `
<div
  className="Pinpoint Copyright "
>
  <a
    className="Pinpoint Logo sm "
    href="https://pinpoint.com"
  >
    <img
      className="image"
      src="https://file.pinpoint.com/1fcde4196a4c70a8a86f0ce4af53f2a5;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png"
    />
  </a>
  <span
    className="text"
  >
    Copyright © 
    2021 by Pinpoint Software, Inc.
  </span>
</div>
`;

exports[`Test with logo 1`] = `
<div
  className="Pinpoint Copyright "
>
  <span
    className="Pinpoint Logo sm  "
  >
    <img
      className="image"
      src="https://file.pinpoint.com/1fcde4196a4c70a8a86f0ce4af53f2a5;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png"
    />
  </span>
  <span
    className="text"
  >
    Copyright © 
    2021 by Pinpoint Software, Inc.
  </span>
</div>
`;

exports[`Test without logo 1`] = `
<div
  className="Pinpoint Copyright "
>
  <span
    className="text"
  >
    Copyright © 
    2021 by Pinpoint Software, Inc.
  </span>
</div>
`;
